### PR TITLE
Initial UI with selectable load clients, displayed load client status…

### PR DESCRIPTION
**Pull request by abassey2**
This gives the initial UI for load client status and run. It has selectable load clients for runs, displayed load client statuses, and a popup when you click the status button that gives the corresponding load client details.

Closes: https://github.com/retaildevcrews/wcnp/issues/226

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -1,25 +1,73 @@
-import logo from './logo.svg';
```
Comment (joheec) [inline client/src/App.js:1]: If this file isn't being used anywhere else, it can be deleted from the repo

File: client/src/App.js [hunk 2]
```diff
@@ -1,25 +1,73 @@
...
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+      <LoadClientContext.Provider 
+        value={{ loadclients, currClientDetails, setCurrClientDetails, handleOpen, handleClose }}>
```
Comment (joheec) [inline client/src/App.js]: Thoughts on having loadClients as part of the context and the state related things passed as props.

File: client/src/LoadClientDetails/index.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import React, { useContext } from "react";
+import LoadClientContext from "../LoadClientContext";
+
+import "./styles.css"
+ 
+const LoadClientDetails = props => {
```
Comment (joheec) [inline client/src/LoadClientDetails/index.js]: props parameter doesn't seem to be used here

File: client/src/LoadClientDetails/index.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  const { loadclients, currClientDetails, handleClose } = useContext(LoadClientContext);
+
```
Comment (joheec) [inline client/src/LoadClientDetails/index.js]: To cut back on code, can create a variable here and reference its properties in the JSX
const details = loadClients[currentClientDetails]

```<h1>{details.name}</h1>```

File: client/src/LoadClientDetails/styles.css [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    height: auto;
+    max-height: 70vh;
+    background: #fff;
+    border-radius: 4px;
+    padding: 20px;
+    border: 5px solid #000;
+    overflow: auto;
+  }
+  .close-icon {
+    content: 'x';
```
Comment (joheec) [inline client/src/LoadClientDetails/styles.css]: I noticed there is an x between the spans in LoadClientDetails/index.js. What happens with this content: 'x'?

File: client/src/LoadClients/index.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
+import { useContext, useState } from 'react';
+import LoadClientContext from '../LoadClientContext';
+
+
+import "./styles.css"
+
+const LoadClients = (props) => {
```
Comment (joheec) [inline client/src/LoadClients/index.js]: this component doesn't seem to be using props

File: client/src/LoadClients/index.js [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+  const { loadclients, handleOpen} = useContext(LoadClientContext)
+  const [clients, setClients] = useState(loadclients);
+
+  function handleToggleSelected(id) {
```
Comment (joheec) [inline client/src/LoadClients/index.js]: Just thoughts on another way to handle this. Not a fan of the index, but maybe you have some ideas to improve it.

```javascript
const updatedLoadClientIndex = clients.findIndex(lc => lc.id === id);
setClients({
    ...clients.slice(0, updatedLoadClientIndex),
    {
        ...clients[updatedLoadClientIndex],
        isExecute: !clients[updatedLoadClientIndex],
    }
    ...clients.slice(updatedLoadClientIndex + 1)
});
```

Comment (joheec) [reply client/src/LoadClients/index.js]: maybe....
```javascript
const updatedLoadClient = {
    ...clients[updatedLoadClientIndex],
    isExecute: !clients[updatedLoadClientIndex],
};
```

**Review by joheec — CHANGES_REQUESTED** [review]

File: client/src/App.js (hunk 2 above)
Comment (abassey2) [reply client/src/App.js]: That makes sense to me as the other ones are only uses in specific files.

File: client/src/LoadClientDetails/styles.css (hunk 5 above)
Comment (abassey2) [reply client/src/LoadClientDetails/styles.css]: looking at it again it seems unnecessary

**Review by joheec — APPROVED** [review]
Looks good!

**Pull request merged**